### PR TITLE
IPFS workflow new style

### PR DESCRIPTION
**Pull request by karmanyaahm**
@p1gp1g what do you think about this? In this version, the ssh key has no permissions (no filesystem access, no command execution) other than to connect to 'localhost:5001' on the server, which is the IPFS api.

### CODE REVIEW

**Comment by karmanyaahm** [comment]
This is what I added to sshd config on the server

```conf
        AuthorizedKeysFile .ssh/authorized_keys /etc/ssh/%u.authkeys
        Match User nobody                               
           AllowTcpForwarding local                     
           AllowAgentForwarding no                      
           X11Forwarding no                             
           PermitTunnel no                              
           PermitOpen localhost:5001 ## ipfs api
           ForceCommand echo 'This account can only be used for Tunneling'
```
and `/etc/ssh/nobody.authkeys` contains the public key for this (GH actions) private key

**Comment by karmanyaahm** [comment]
Please merge if you think this is good

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -39,17 +39,23 @@ jobs:
...
-      - name: deploy-upload
+      - name: Connect to ssh in BG
+        timeout-minutes: 2
         run: | 
           echo "${{ secrets.SSHKEY }}" > ../privkey
           chmod 600 ../privkey
-          ssh -o StrictHostKeyChecking=no ubuntu@unifiedpush.org -i ../privkey rm -rf /tmp/action-sftp-upload
-          ssh -o StrictHostKeyChecking=no ubuntu@unifiedpush.org -i ../privkey mkdir -p /tmp/action-sftp-upload
-          scp -r -i ../privkey ./public/* ubuntu@unifiedpush.org:/tmp/action-sftp-upload
-          HASH=$(ssh -o StrictHostKeyChecking=no ubuntu@unifiedpush.org -i ../privkey sudo -u ipfs ipfs add --recursive --pin=false --quieter /tmp/action-sftp-upload)
```
Comment (p1gp1g) [inline .github/workflows/main.yml:49]: I guess it won't update the DNS record now ^^

**Review by p1gp1g — COMMENTED** [review]
Nice way to do it !

File: .github/workflows/main.yml (hunk 1 above)
Comment (karmanyaahm) [reply .github/workflows/main.yml:49]: The IPFS action in the next line handles that

Comment (p1gp1g) [reply .github/workflows/main.yml:49]: Great :)

**Pull request merged**